symengine/add.cpp

```diff
@@ -324,10 +324,18 @@ void Add::as_two_terms(const Ptr<RCP<const Basic>> &a,
 vec_basic Add::get_args() const
 {
     vec_basic args;
-    if (not coef_->is_exact_zero())
+    if (not coef_->is_exact_zero()) {
+        args.reserve(dict_.size() + 1);
         args.push_back(coef_);
+    } else {
+        args.reserve(dict_.size());
+    }
     for (const auto &p : dict_) {
-        args.push_back(Add::from_dict(zero, {{p.first, p.second}}));
+        if (eq(*p.second, *one)) {
+            args.push_back(p.first);
+        } else {
+            args.push_back(Add::from_dict(zero, {{p.first, p.second}}));
+        }
```

Comment:
Is this `eq(*p.second, *one)` optimization specific to `Add`, or is this something that should be moved to `Add::from_dict`?

Reply:
It's added in `Add::from_dict`, but `Add::from_dict` checks lots of things in addition to that. For this case, we can bypass other checks.